xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h

```diff
@@ -203,14 +203,17 @@ namespace ADDON
      * @brief Add a message to XBMC's log.
      * @param loglevel The log level of the message.
      * @param format The format of the message to pass to XBMC.
+     * @note This method uses limited buffer (16k) for the formatted output.
+     * So data, which will not fit into it, will be silently discarded.
      */
     void Log(const addon_log_t loglevel, const char *format, ... )
     {
       char buffer[16384];
       va_list args;
       va_start (args, format);
-      vsprintf (buffer, format, args);
+      vsnprintf (buffer, sizeof (buffer), format, args);
```

Comment:
Is `vsnprintf()` like `strncpy()`? If so, this should be `vsnprintf(buffer, sizeof(buffer) - 1, format, args);`

Reply:
right, but we can gain a 0.00001% speedup with the lower bound :-D. I'll change the commit to make everyone happy.

Reply:
... with the assumption that `sizeof(buffer) - 1` is evaluated in compile time